### DISCUSSION TITLE
Misleading rule description for e.g. BR-BT-00512-0011

### DISCUSSION THREAD

**Discussion by rkottmann** [OP-TED/eForms-SDK#867] [Q&A] [answered]
Hi,

This 

```
<assert id="BR-BT-00512-0011" role="ERROR" diagnostics="ND-Company_BT-512-Organization-Company" 
   test="
count(cac:PostalAddress/cbc:PostalZone) &gt; 0 
     or 
not(cac:PostalAddress/cac:Country/cbc:IdentificationCode/normalize-space(text()) =
 ('AFG','ALA','ALB','AND','ARG','ARM','AUS','AUT','AZE','BEL','BGD','BGR','BHR','BIH','BLM','BLR','BMU','BRA','BRB','BRN','BTN','CAN','CHE','CHL','CHN','COL','CPT','CPV','CRI','CUB','CYM','CYP','CZE','DEU','DJI','DNK','DOM','DZA','ECU','EGY','ESP','EST','ETH','FIN','FRA','FRO','FSM','GBR','GEO','GGY','GIN','GNB','GRC','GRL','GTM','GUM','HND','HRV','HTI','HUN','IDN','IMN','IND','IRN','IRQ','ISL','ISR','ITA','JEY','JOR','JPN','KAZ','KEN','KGZ','KHM','KIR','KOR','KWT','LAO','LBN','LIE','LKA','LSO','LTU','LUX','LVA','MAF','MAR','MCO','MDA','MDG','MDV','MEX','MHL','MLT','MMR','MNE','MNG','MOZ','MSR','MUS','MWI','MYS','NAM','NCL','NER','NGA','NIC','NLD','NOR','NPL','NRU','NZL','OMN','PAK','PER','PHL','PNG','POL','PRI','PRT','PRY','PSE','PYF','ROU','RUS','SAU','SDN','SEN','SGP','SJM','SLV','SMR','SPM','SRB','SVK','SVN','SWE','SWZ','THA','TJK','TKM','TLS','TTO','TUN','TUR','UKR','URY','USA','UZB','VCT','VEN','VGB','VIR','VNM','WLF','ZAF'))">rule|text|BR-BT-00512-0011</assert>
```

implements via boolean logic an if-then statement which can be phrased as `If an Organization-Company is in a country having postal zones then cac:PostalAddress/cbc:PostalZone must exist` 

Hence the message

```
<entry key="rule|text|BR-BT-00512-0011">BT-512-Organization-Company is mandatory for a notice with subtype T01 under the condition: Organisation country (BT-514-Organization-Company) is not a country with post codes</entry>
```

is misleading and might be corrected to

```
<entry key="rule|text|BR-BT-00512-0011">BT-512-Organization-Company is mandatory for a notice with subtype T01 under the condition: Organisation country (BT-514-Organization-Company) is IN  a country with post codes</entry>
```

Additionally, IMHO there can only be one postal code (here also already enforced by XSD) hence `count(cac:PostalAddress/cbc:PostalZone) = 1` might be more precise and better revealing intent.

thx for considering.

ciao,

renzo

**Comment by YvesJo** [comment, marked as answer]
Hi,
Could you indicate the SDK version involved?
For the rule in question, I'm finding the following label:

> 'Postcode' (BT-512-Organization-Company) is mandatory in notice type 'T01 – Prior information notice for public service contract' except when: 'Country' (BT-514-Organization-Company) does not have postcodes

Be aware that, given the number of rules, a canvas and specific expressions are predefined. Existence of a field does not preclude a single occurrence of that field; occurrences are addressed by the schema, also existence is here sufficient.

**Reply by rkottmann** [reply]
Version 1.10.0 (sry for having missed this)

**Reply by YvesJo** [reply]
Hi,
For SDK 1.10.0 (and SDK 1.10.1) I find the following entry
<entry key="rule|text|BR-BT-00512-0011">'Postcode' (BT-512-Organization-Company) is mandatory in notice type 'T01 – Prior information notice for public service contract' **except under following condition: 'Country' (BT-514-Organization-Company) does not have postcodes**</entry>
So, most likely the referred text comes from another version or from a release candidate
